solvers/cuml.py

```diff
@@ -0,0 +1,49 @@
+from benchopt import BaseSolver, safe_import_context
+from benchopt.utils.sys_info import _get_cuda_version
+
+with safe_import_context() as import_ctx:
+    import cudf
+    import numpy as np
+    from cuml.linear_model import LogisticRegression
+
+
+class Solver(BaseSolver):
+    name = "cuml"
+
+    cuda_version = _get_cuda_version()
+    if cuda_version is not None:
+        install_cmd = "conda"
+        cuda_version = cuda_version.split("cuda_", 1)[1][:4]
+        requirements = [
+            "rapidsai::rapids",
+            f"nvidia::cudatoolkit={cuda_version}",
+            "dask-sql",
+        ]
```

Comment:
I think it is a bit cleaner if `install_cmd` and `requirements` are always defined.

Also add a more direct error for why the solver is not working.

```suggestion
cuda_version = _get_cuda_version()
if cuda_version is not None:
    cuda_version = cuda_version.split("cuda_", 1)[1][:4]

with safe_import_context() as import_ctx:
    if cuda_version is None:
        raise ImportError("cuml solver needs a nvidia GPU to run")

    import cudf
    import numpy as np
    from cuml.linear_model import LogisticRegression


class Solver(BaseSolver):
    name = "cuml"

    install_cmd = "conda"
    requirements = [
        "rapidsai::rapids",
        f"nvidia::cudatoolkit={cuda_version}",
        "dask-sql",
    ] if cuda_version is not None else []
```